compose/service.py

```diff
@@ -241,29 +245,15 @@ def recreate_container(self, container, **override_options):
             else:
                 raise
 
-        intermediate_container = Container.create(
-            self.client,
-            image=container.image,
-            entrypoint=['/bin/echo'],
-            command=[],
-            detach=True,
-            host_config=create_host_config(volumes_from=[container.id]),
-        )
-        intermediate_container.start()
-        intermediate_container.wait()
-        container.remove()
-
-        options = dict(override_options)
+        # Use a hopefully unique container name by prepending the short id
+        self.client.rename(container.id, container.short_id + container.name)
```

Comment:
This is also possible right now, isn't it? If Compose crashes after creating the intermediate container but before creating the new one, it won't recreate.

It's not ideal, but at least it isn't a regression ^_^